src/ovms.h

```diff
@@ -171,37 +290,143 @@ OVMS_Status* OVMS_ModelsSettingsSetConfigPath(OVMS_ModelsSettings* settings,
+//
+// \param request The request object
+// \param inputName The name of the input with data to be removed
+// \param data The data of the input
+// \param byteSize The byte size of the data
+// \param bufferType The buffer type of the data
+// \param deviceId The device id of the data memory buffer
+// \return OVMS_Status object in case of failure
+OVMS_Status* OVMS_InferenceRequestInputSetData(OVMS_InferenceRequest* request, const char* inputName, const void* data, size_t byteSize, OVMS_BufferType bufferType, uint32_t deviceId);
+
+// Remove the data of the input buffer
+//
+// \param request The request object
+// \param inputName The name of the input with data to be removed
+// \return OVMS_Status object in case of failure
 OVMS_Status* OVMS_InferenceRequestInputRemoveData(OVMS_InferenceRequest* request, const char* inputName);
-OVMS_Status* OVMS_InferenceRequestRemoveInput(OVMS_InferenceRequest* request, const char* inputName);  // this will allow for reuse of request but with different input data
+
+// Remove input from the request
+//
+// \param request The request object
+// \param inputName The name of the input to be removed
+// \return OVMS_Status object in case of failure
+OVMS_Status* OVMS_InferenceRequestRemoveInput(OVMS_InferenceRequest* request, const char* inputName);
+
+//
+// Add parameter to the request
+//
+// \param request The request object
+// \param parameterName The name of the parameter to be added
+// \param datatype The request object
+// \param data The data representing parameter value
+// \param byteSize The byte size of the added parameter value
+// \return OVMS_Status object in case of failure
 OVMS_Status* OVMS_InferenceRequestAddParameter(OVMS_InferenceRequest* request, const char* parameterName, OVMS_DataType datatype, const void* data, size_t byteSize);
+
+// Remove parameter from the inference request
+//
+// \param request The request object
+// \param parameterName The name of the parameter to be removed
+// \return OVMS_Status object in case of failure
 OVMS_Status* OVMS_InferenceRequestRemoveParameter(OVMS_InferenceRequest* request, const char* parameterName);
 
 // OVMS_Inference Response
+//
+// Get the number of outputs in the response
+//
+// \param response The response object
+// \param count The value to be set
+// \return OVMS_Status object in case of failure
 OVMS_Status* OVMS_InferenceResponseGetOutputCount(OVMS_InferenceResponse* response, uint32_t* count);
-OVMS_Status* OVMS_InferenceResponseGetOutput(OVMS_InferenceResponse* response, uint32_t id, const char** name, OVMS_DataType* datatype, const uint64_t** shape, uint32_t* dimCount, const void** data, size_t* bytesize, OVMS_BufferType* bufferType, uint32_t* deviceId);
+
+// Get all information about an output from the response.
+//
+// \param response The response object
+// \param id The id of the output
+// \param name The name of the output
+// \param datatype The data type of the output
+// \param shape The shape of the output
```

Comment:
Should we distinguish `**` from `*` in the description? Some `in`/`out` ? So end user is sure what is user input and what is user output:
Example: response/id is input
all the rest is output in this case

As you can see user cannot deduce from **/*, for example datatype

Suggestion
```
// \param response [in] The response object
// \param id [in] The id of the output
// \param name [out] The name of the output
// \param datatype [out] The data type of the output
...
```

Reply:
The name of the function suggest that we are getting something. See also the first comment about this function,

Reply:
Yes, we get something, but not all params are for "getting something" - first 2 params are for providing something

Reply:
Added clearer description in the comment